src/identify/index.js

```diff
@@ -258,17 +306,73 @@ class IdentifyService {
       return log.error('received invalid message', err)
     }
 
-    // Update peers data in PeerStore
     const id = connection.remotePeer
+
+    // Legacy
+    if (!message.signedPeerRecord) {
+      try {
+        this.peerStore.addressBook.set(id, message.listenAddrs.map((addr) => multiaddr(addr)))
+      } catch (err) {
+        return log.error('received invalid listen addrs', err)
+      }
+
+      // Update the protocols
+      this.peerStore.protoBook.set(id, message.protocols)
+
+      return
+    }
```

Comment:
Again, I'd try the record first to allow fallback to happen if it exists but can't be unmarshalled.